### PR TITLE
DAG-1153: Add method to check if a task was a failure or not

### PR DESCRIPTION
**Pull request by zamj**

### CODE REVIEW

**Review by dbradf — COMMENTED** [review]
I'm guessing this is not going to do what you actually want. The `status` field in evergreen is pretty overloaded. As a result, both timeouts and system failures, as well as test failures, have a status of `failed`. If you are just concerned with test failures, which I think you are, you will need to check the `status_details` field to determine the type of failure that is being seen. You can see the `is_timeout` and `is_system_failure` methods for examples.

With all that in mind, we also might want a more specific name than `is_failure` to not confuse this with evergreen terminology. Maybe `is_test_failure`?

Also, we should probably add some tests.

**Comment by zamj** [comment]
> I'm guessing this is not going to do what you actually want. The `status` field in evergreen is pretty overloaded. As a result, both timeouts and system failures, as well as test failures, have a status of `failed`. If you are just concerned with test failures, which I think you are, you will need to check the `status_details` field to determine the type of failure that is being seen. You can see the `is_timeout` and `is_system_failure` methods for examples.
> 
> With all that in mind, we also might want a more specific name than `is_failure` to not confuse this with evergreen terminology. Maybe `is_test_failure`?
> 
> Also, we should probably add some tests.

Interesting, didn't realize failure was that general in the evg api. You're correct, I'm trying to check if a task failed due to a test failure or not. Thanks for letting me know, I've updated it.

File: src/evergreen/task.py [hunk 1]
```diff
@@ -18,6 +18,7 @@
 EVG_SUCCESS_STATUS = "success"
 EVG_SYSTEM_FAILURE_STATUS = "system"
 EVG_UNDISPATCHED_STATUS = "undispatched"
+EVG_TEST_STATUS = "test"
```
Comment (dbradf) [inline src/evergreen/task.py:21]: Could we rename this to `EVG_TEST_STATUS_TYPE`? Since it is a status type and not a status.

**Review by dbradf — APPROVED** [review]

**Comment by zamj** [comment]
evergreen merge

**Pull request merged**